src/sql/workbench/services/connection/browser/connectionDialogWidget.ts

```diff
@@ -616,7 +647,7 @@ export class ConnectionDialogWidget extends Modal implements IViewPaneContainer
 		const disposable = combinedDisposable(pane, paneStyler);
 		const paneItem = { pane, disposable };
 		treeView.onDidChangeSelection(e => {
-			if (e.length > 0 && e[0].payload) {
+			if (e.length > 0 && e[0].payload instanceof ConnectionProfile) {
```

Comment:
Is the element guaranteed to have a payload? I think only the extensions have them. If the payload isn't there, this would break then

Reply:
`undefined instanceOf class` is valid and will return false

Reply:
https://repl.it/languages/typescript#index.ts